File: program/program/implement-linear-search/implement_linear_search.pl
#!/usr/bin/perl

use strict;
use warnings;

sub linear_search {
    my ($list_ref, $value) = @_;
    my @list = @$list_ref;

    for my $i (0 .. $#list) {
        if ($list[$i] == $value) {
            return $i;
        }
    }

    return -1; # Not found
}

# Input the list from the user
print "Enter numbers separated by spaces: ";
my $input = <STDIN>;
chomp($input);
my @list = split(/\s+/, $input);

# Input the value to search for
print "Enter the value to search for: ";
my $value = <STDIN>;
chomp($value);

my $index = linear_search(\@list, $value);

if ($index != -1) {
    print "Output: $index\n";
} else {
    print "Value not found in the list.\n";
}

File: program/program/implement-linear-search/implement_linear_search.php
<?php
function linear_search($list, $value) {
    $length = count($list);
    for ($i = 0; $i < $length; $i++) {
        if ($list[$i] == $value) {
            return $i; // Found the value, return its location/index
        }
    }
    return -1; // Value not found in the list
}

// Prompt the user to enter a list of numbers separated by spaces
echo "Enter a list of numbers separated by spaces: ";
$input = trim(fgets(STDIN));
$list = explode(' ', $input);

// Prompt the user to enter the value to search for
echo "Enter the value to search for: ";
$searchValue = trim(fgets(STDIN));

$result = linear_search($list, $searchValue);

if ($result != -1) {
    echo "Value $searchValue found at index $result";
} else {
    echo "Value $searchValue not found in the list";
}
?>
